### PR TITLE
Pass epoch length in tests setup

### PR DESCRIPTION
**Pull request by yordanmadzhunkov**
Fix `KeyValueRuntime` to take epoch length as parameter. It caused some bugs with epochs with different length then 5
Replace direct call of Utc::now with Clock::utc() so it can be mocked in tests

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -4257,9 +4257,10 @@ impl<'a> ChainUpdate<'a> {
         let prev_header = self.get_previous_header(header)?.clone();
 
         // Check that epoch_id in the header does match epoch given previous header (only if previous header is present).
-        if &self.runtime_adapter.get_epoch_id_from_prev_block(header.prev_hash())?
-            != header.epoch_id()
-        {
+        let epoch_id_from_prev_block =
```
Comment (mm-near) [inline chain/chain/src/chain.rs:4260]: this doesn't seem to be related to the PR title. (or is it just a drive-by refactor? ;-)

File: chain/chain/src/tests/simple_chain.rs [hunk 2]
```diff
@@ -36,6 +36,7 @@ fn build_chain() {
     let _mock_clock_guard = MockClockGuard::default();
     for i in 0..5 {
         Clock::add_utc(chrono::Utc.ymd(2020, 10, 1).and_hms_milli(0, 0, 3, 444 + i));
+        Clock::add_utc(chrono::Utc.ymd(2020, 10, 1).and_hms_milli(0, 0, 3, 444 + i));
```
Comment (mm-near) [inline chain/chain/src/tests/simple_chain.rs:39]: can you comment why are we adding it two times?

Comment (mm-near) [reply chain/chain/src/tests/simple_chain.rs:39]: is the second call here due to 'validate_header' call?

File: chain/chain/src/tests/simple_chain.rs [hunk 3]
```diff
@@ -62,17 +63,17 @@ fn build_chain() {
...
     let count_instant = Clock::instant_call_count();
     let count_utc = Clock::utc_call_count();
-    assert_eq!(count_utc, 5);
+    assert_eq!(count_utc, 9);
     assert_eq!(count_instant, 0);
     #[cfg(feature = "nightly_protocol")]
     assert_eq!(
         chain.head().unwrap().last_block_hash,
-        CryptoHash::from_str("BNap11nsM7PEqYQertYU487g7gkCteQ8Fmee6QfyRdVQ").unwrap()
+        CryptoHash::from_str("JDNJcU8dbsY18Trzbts9gSphtQqw9C5fx1Abg3SzDdRZ").unwrap()
```
Comment (mm-near) [inline chain/chain/src/tests/simple_chain.rs:71]: can you explain why the header has changed?

**Review by mm-near — CHANGES_REQUESTED** [review]

File: chain/chain/src/chain.rs (hunk 1 above)
Comment (yordanmadzhunkov) [reply chain/chain/src/chain.rs:4260]: yes, it's a drive by refactor. It was easier to debug in this version

File: chain/chain/src/tests/simple_chain.rs (hunk 2 above)
Comment (yordanmadzhunkov) [reply chain/chain/src/tests/simple_chain.rs:39]: yes, the validate header consumes 4 more entries from the clock in this test scenario.

File: chain/chain/src/tests/simple_chain.rs (hunk 3 above)
Comment (yordanmadzhunkov) [reply chain/chain/src/tests/simple_chain.rs:71]: I will revert them.

**Review by yordanmadzhunkov — COMMENTED** [review]
try to merge

**Review by mm-near — APPROVED** [review]
